### ISSUE TITLE
about the pretrained model

### ISSUE THREAD

**Issue by dami23** [mikittt/Human-Friendly-REG#1]
@mikittt Dear Mikihiro, thank you for the share of your work. I have a question about the uploaded models, sp.h5, splm.h5, and spve.h5 are for RefGTA, it is right? could you also please provide the generation models from Refcoco, Refcoco+ and Refcocog? Thank you a lot.

**Comment by mikittt** [comment]
You are right.  I'll inform you here after uploading those pre-trained models, so please wait for a while.

**Comment by dami23** [comment]
@mikittt thank you for your reply and share. Thank you again!

**Comment by mikittt** [comment]
I uploaded the pre-processed data and pre-trained models using the current code. The path settings are written in config.py, so please change it according to your environment.

**Comment by dami23** [comment]
@mikittt  sincerely thank you for your kindness help and generous. Thanks a lot!!

**Comment by dami23** [comment]
@mikittt sorry to disturb you again. I found a question when I run the code ''scripts/extract_target_feats.py". The generated sp_ann_feats.npy is so big, it will cause memory error, how did you deal with this error? Thank you!

**Comment by mikittt** [comment]
I'm sorry, but the training codes only work on a machine with enough memory. 
It would be better to save each feature separately to deal with the problem, but I've not supported it.

**Comment by dami23** [comment]
@mikittt I found mismatch errors when loading the trained model. refococ: cocolm.h5  Shape mismatch. Old shape: (1992, 512), new shape: (2264, 512); refococ+: coco+lm.h5 Shape mismatch. Old shape: (2627, 512), new shape: (2914, 512).
Could you please give a cue to tackle with the errors? Thank you for your help!!

**Comment by dami23** [comment]
@mikittt Excuse me, I sorry to disturb you again. The generated expressions by the uploaded models are weird, such as refcoco {u'sent': u'doggy written skating stoplight', u'ref_id': 49834}, the same situation in RefGTA. Could you please give me some advice to tackle this problem? Thank you!

**Comment by mikittt** [comment]
Will you check whether you're using the 'data.json' which I uploaded? If you're using the 'data.json' which is generated by yourself, the indices of the vocabulary get different. If it's not the case, please tell me again.

**Comment by dami23** [comment]
@mikittt Hi, Mikihiro. You are right, the problem is the self-generated dataset.h5 and dataset.json. I followed the instructions in ReadMe to generated the required files. I have checked the code prepro.py and did not find any bug in prepro.py. However, I cannot figure out the cause, could you please give me some hints? sincerely thank you for your reply and help!

**Comment by mikittt** [comment]
I'm sorry for the lack of explanation. The indices corresponding to words change every time you run prepro.py. The pre-trained models are trained based on the dataset.h5 and dataset.json I uploaded, so please use them when you use these models. 
For example, if the model I uploaded trained on RefCOCO outputs index 2245, it means the model generates 'man' as you can see by ``` json.load(open('data.json','r'))['ix_to_word']['2245']```, but a different word will be returned when you use self-generated 'data.json'.
